packages/webpack-cli/src/plugins/dotenv-webpack-plugin.ts

```diff
@@ -0,0 +1,179 @@
+import dotenv from "dotenv-defaults";
+import { Compiler, DefinePlugin } from "webpack";
+
+interface EnvVariables {
+  [key: string]: string;
+}
+
+interface Logger {
+  info(message: string): void;
+  warn(message: string): void;
+  error(message: string): void;
+}
+
+interface DotenvConfig {
+  paths?: string[];
+  prefixes?: string[];
+  systemvars?: boolean;
+  allowEmptyValues?: boolean;
+  expand?: boolean;
+  ignoreStub?: boolean;
+  defaults?: boolean | string;
+}
+
+const interpolate = (env: string, vars: EnvVariables): string => {
+  const matches = env.match(/\$([a-zA-Z0-9_]+)|\${([a-zA-Z0-9_]+)}/g) || [];
+  matches.forEach((match) => {
+    env = env.replace(match, interpolate(vars[match.replace(/\$|{|}/g, "")] || "", vars));
+  });
+  return env;
+};
+
+const isMainThreadElectron = (target: string | undefined): boolean =>
+  !!target && target.startsWith("electron") && target.endsWith("main");
+
+export class Dotenv {
+  private logger!: Logger;
+  private config: DotenvConfig;
+  private cache: Record<string, string>;
+  private inputFileSystem: any;
+
+  constructor(config: DotenvConfig = {}) {
+    this.config = {
+      paths: ["./.env"],
```

Comment:
Default values should be other

Reply:
can you please elaborate on this, like do u mean the ./.env should be ./.env.local or something

Reply:
it should be:
```
.env.$(NODE_ENV).local
.env.local (Not checked when NODE_ENV is test.)
.env.$(NODE_ENV)
.env
```

NODE_ENV is `process.env.NODE_ENV`, if the environment variable NODE_ENV is unassigned it is `compilation.option.mode`, so:

JUST pseudo code:
```js
const nodeEnv = process.env.NODE_ENV || compilation.options.mode;
const possiblePaths = [".env.[node_env].local"]
  .concat(nodeEnv === "test" ? [] : [".env.local "])
  .concat(".env.[node_env]", ".env")
].map(item => item.replace("[node_env]", nodeEnv))
```